swap/swap.go

```diff
@@ -145,20 +144,42 @@ func New(logpath string, dbPath string, prvkey *ecdsa.PrivateKey, backendURL str
 	if err != nil {
 		return nil, fmt.Errorf("swap init error: %s", err)
 	}
-	if disconnectThreshold <= paymentThreshold {
-		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", disconnectThreshold, paymentThreshold)
+	if params.DisconnectThreshold <= params.PaymentThreshold {
+		return nil, fmt.Errorf("swap init error: disconnect threshold lower or at payment threshold. DisconnectThreshold: %d, PaymentThreshold: %d", params.DisconnectThreshold, params.PaymentThreshold)
 	}
 	backend, err := ethclient.Dial(backendURL)
 	if err != nil {
 		return nil, fmt.Errorf("swap init error: error connecting to Ethereum API %s: %s", backendURL, err)
 	}
-	return new(
-		logpath,
+
+	if params != nil && params.InitialDepositAmount == 0 {
+		// need to prompt user for initial deposit amount
+		// if 0, can not cash in cheques
+		prompter := console.Stdin
+
+		// we may not need this check, and we could maybe even get rid of this constant completely
+		// ask user for input
+		input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment")
```

Comment:
Please add a ` ` or `: ` here. Right now the user input comes directly after the output without any spacing.
e.g. `Please provide the amount in Wei which will deposited to your chequebook upon deployment4353`

Reply:
Great catch! Will do, thanks.

You obviously caught this because you went, compiled and ran it. Awesome attitude. Thank you very much.